src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -96,7 +106,12 @@ public void deleteWorkspace(UUID id, AuthenticatedUserRequest userReq) {
                 null, // Delete does not have a useful request body
                 userReq)
             .addParameter(WorkspaceFlightMapKeys.WORKSPACE_ID, id);
-    deleteJob.submitAndWait(null);
+    try {
+      deleteJob.submitAndWait(null);
+    } catch (DuplicateFlightIdSubmittedException ex) {
+      // FlightIDs used here are random UUIDs, so a collision is purely bad luck.
+      throw new InternalStairwayException(ex);
+    }
```

Comment:
if this is more common than not, it seems like our jobservice isn't pulling enough weight for us. Leaning more towards separate functions or parameter for allowing/failing duplicate job ids.

Reply:
I'm adding a param here, I'd prefer not to break out another function when we already have both `submit` and `submitAndWait`. 

We could also consider a small refactor of JobBuilder to accept jobId as optional on new jobs and generate an ID if one isn't provided. That puts flightID generation all in one place instead of generating random UUIDs across WM, and also lets us make the distinction of "provided duplicate flightID -> idempotent behavior", "generated duplicate flightID -> (rare) error".

I think that's getting a little out of scope and this PR is already large though, so I'll leave that as a separate issue.

Reply:
I like the idea of moving it into JobBuilder. Another PR sgtm.
(fwiw, another thing that's bothered me about the current JobBuilder API is that it wraps setting up the Flight input parameters so that the code that relies on the input parameters (the flight & its step) is in a totally separate place than the code that sets up the job and sets the parameters. Totally unrelated to what's here, but one of my pet peeves if/when you do take a look)